tests/Testing/TurboStreamMatcherTest.php

```diff
@@ -37,30 +43,37 @@ protected function setUp(): void
     /** @test */
     public function converts_streams_to_collections()
     {
-        $this->assertCount(3, $this->streams);
+        $this->assertCount(4, $this->streams);
     }
 
     /** @test */
     public function filters_by_attributes()
     {
         $appends = $this->streams->filter(fn (TurboStreamMatcher $matcher) => (
-            $matcher->where('action', 'append')->matches()
+            (clone $matcher)->where('action', 'append')->matches()
```

Comment:
Since the same `$matcher` is re-used for all the tests here, and it's mutable, all previous `where`'s apply to future tests. This wasn't an issue before, since each test fully override all `wheres`. The new case doesn't override `target`, since we don't expect it to be there.

The cleanest solution would probably be to make sure each filter gets a clean `$matcher` instance, but we didn't want to balloon this PR too much and this seemed like a good solution.

Reply:
I solved it by cloning the matcher whenever we call the `->where()` or `->see()` methods (see [here](https://github.com/tonysm/turbo-laravel/pull/87/commits/6fa7f23d809e251dac2166d79df2e765dd8e62e4)). I also had to make sure that the `hasTurboStream()` was dealing with this matcher clone so the assertion messages worked (see [here](https://github.com/tonysm/turbo-laravel/pull/87/commits/962873af5a5d95900a0a2cf86e6f8dea046659b4) and [here](https://github.com/tonysm/turbo-laravel/pull/87/commits/b9d9c21046c39f899d8e5c73c9c52b6eef636d5e))